cli/commands/helpers/flags.go

```diff
@@ -75,10 +79,36 @@ func GetChangedStringValueFlag(name string, flagset *pflag.FlagSet) string {
 	if value, err := flagset.GetString(name); err == nil {
 		return value
 	}
-
 	return ""
 }
 
+// EnvHasChanged determines if the user has set the value of an environment variable or flag
+// or left it to default
+func EnvHasChanged(name string, v *viper.Viper) bool {
+	return v.IsSet(name)
+}
+
+// GetChangedStringValueEnv returns the value of an environment variable or flag that has been explicitly
+// changed by the user, and not left to default
+func GetChangedStringValueEnv(name string, v *viper.Viper) string {
+	if !EnvHasChanged(name, v) {
+		return ""
+	}
+
+	return v.GetString(name)
+}
+
+// GetChangedStringValueViper returns the value of an environment variable or flag that has been explicitly
+// changed by the user, and not left to default, from a Cobra FlagSet.
+func GetChangedStringValueViper(name string, flagset *pflag.FlagSet) string {
```

Comment:
@ccressent I ran out of inspiration for the name. If you have a better suggestion please feel free to share :)